Makefile

```diff
@@ -691,3 +691,6 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
 
 undeploy: ## Undeploy controller from the K8s cluster specified in $KUBECONFIG.
 	$(KUSTOMIZE) build config/default | kubectl delete -f -
+
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
```

Comment:
That looks like black magic 🧙 🪄 Any chance to make this clear for the reader?

I wonder if there's benefit for users of this repo to have this type of `make` target 🤔 Wouldn't it be better if we make a better job of describing what should be run in [`README.md`](https://github.com/Kong/kubernetes-ingress-controller/blob/main/README.md) and [`TESTING.md`](https://github.com/Kong/kubernetes-ingress-controller/blob/main/TESTING.md)? Not all targets make sense to be run by themselves, e.g. the ones that install the tools should be run by the steps that use them

Reply:
Modifying the document is a very intuitive way :+1: , but it requires developers to be willing to read the document in its entirety. 

Alternatively, we can also add some rules to only display certain entry points that we want developers to see.